### ISSUE TITLE
GitHub Desktop "The name is not activatable" on Arch Linux

### ISSUE THREAD

**Issue by ghost** [shiftkey/desktop#1020]
### The problem

GitHub Desktop is completely unusable on Arch Linux.

Everytime after authorizing the app with my GitHub account the error "The name is not activatable" appears and GitHub Desktop fails to connect to my GitHub account and I cannot access my repositories.

Since there is no error code or anything else that could hint to the cause, the only things I can provide is a screenshot and the production.log inside ~/.config/GitHub Desktop/logs/

### Release version

Version 3.3.8-linux2 (x64)

### Operating system

OS: Arch Linux x86_64 | Kernel: 6.7.8-arch1-1 | DE: KDE Plasma 6.0.1 on Wayland

### Steps to reproduce the behavior

1. Install GitHub Desktop from the AUR: https://aur.archlinux.org/packages/github-desktop
2. Open GitHub Desktop on Arch Linux.
3. Sign in with your GitHub account.
4. Authorize GitHub Desktop in your webbrowser.
5. Let the GitHub website redirect back to the GitHub Desktop application.
6. The error "The name is not activatable" will appear with no further information.

### Log files

[2024-03-08.desktop.production.log](https://github.com/shiftkey/desktop/files/14532464/2024-03-08.desktop.production.log)

### Screenshots

![Screenshot_20240307_183042](https://github.com/shiftkey/desktop/assets/115804951/0d4ef7e9-798d-4a54-aae2-b9719652c630)

**Comment by Zoddhand** [comment]
same issue here.

**Comment by olegovechko** [comment]
I have the same problem.

**Comment by shiftkey** [comment]
This is a generic DBUS message from the backing store we need to persist credentials, and it's really all the context we see from the application side. I'd suggest checking the `gnome-keyring` keychain is setup.

**Comment by ShoreNinth** [comment]
> This is a generic DBUS message from the backing store we need to persist credentials, and it's really all the context we see from the application side. I'd suggest checking the `gnome-keyring` keychain is setup.

Thanks, sir. It's solved after I have `gnome-keyring` installed and set up.

**Comment by BoxedLight** [comment]
> > This is a generic DBUS message from the backing store we need to persist credentials, and it's really all the context we see from the application side. I'd suggest checking the `gnome-keyring` keychain is setup.
> 
> Thanks, sir. It's solved after I have `gnome-keyring` installed and set up.

Yup, can confirm that installing `gnome-keyring` solves the issue.

I guess this issue applies only to those people who dont have Gnome as their desktop environment? I'm not sure since I never used Gnome. Have been a loyal KDE user since day 1.

Would be nice if at least `gnome-keyring` would be added into the PKGBUILD of the github-desktop AUR package as an dependency so it gets installed along with github-desktop.

**Comment by BoxedLight** [comment]
> > This is a generic DBUS message from the backing store we need to persist credentials, and it's really all the context we see from the application side. I'd suggest checking the `gnome-keyring` keychain is setup.
> 
> Thanks, sir. It's solved after I have `gnome-keyring` installed and set up.

Yup, can confirm that installing `gnome-keyring` solves the issue.

I guess this issue only applies to users who don't have Gnome as their desktop environment? I'm not sure since I never used Gnome. Have been a loyal KDE user since day 1.

Would be nice if at least gnome-keyring would be added into the PKGBUILD of the github-desktop AUR package as an dependency so it gets installed along with github-desktop.

**Comment by BoxedLight** [comment]
> > This is a generic DBUS message from the backing store we need to persist credentials, and it's really all the context we see from the application side. I'd suggest checking the `gnome-keyring` keychain is setup.
> 
> Thanks, sir. It's solved after I have `gnome-keyring` installed and set up.

Yup, can confirm that installing `gnome-keyring` solves the issue.

I guess this issue only applies to users who don't have Gnome as their desktop environment? I'm not sure since I never used Gnome. Have been a loyal KDE user since day 1.

Would be nice if at least gnome-keyring would be added into the PKGBUILD of the github-desktop AUR package as a dependency so it gets installed along with github-desktop.

**Comment by ghost** [comment]
> > This is a generic DBUS message from the backing store we need to persist credentials, and it's really all the context we see from the application side. I'd suggest checking the `gnome-keyring` keychain is setup.
> 
> Thanks, sir. It's solved after I have `gnome-keyring` installed and set up.

Yup, can confirm that installing `gnome-keyring` solves the issue.

I guess this issue only applies to users who don't have GNOME as their desktop environment? I'm not sure since I never used GNOME. Have been a loyal KDE user since day 1.

Would be nice if at least gnome-keyring would be added into the PKGBUILD of the github-desktop AUR package as a dependency so it gets installed along with github-desktop.

**Comment by Richardk2n** [comment]
The PKGBUILD has `org.freedesktop.secrets` as a dependency. Which is provided by `gnome-keyring` but also by `kwallet` which works fine for me. At least for continued use. This might be related to initial setup.

**Comment by BoxedLight** [comment]
> The PKGBUILD has `org.freedesktop.secrets` as a dependency. Which is provided by `gnome-keyring` but also by `kwallet` which works fine for me. At least for continued use. This might be related to initial setup.

Nope, I just setup GitHub Desktop on Arch with `gnome-keyring` installed. Worked flawlessly.

But as soon as I uninstalled `gnome-keyring` GitHub Desktop immediately stops working, even if you already set up your account and repositories.

![Screenshot_20240405_180447](https://github.com/shiftkey/desktop/assets/115804951/8e3eda83-7783-4be4-b2cb-f22165777ca5)

**Comment by ghost** [comment]
> The PKGBUILD has `org.freedesktop.secrets` as a dependency. Which is provided by `gnome-keyring` but also by `kwallet` which works fine for me. At least for continued use. This might be related to initial setup.

Nope, I just set up GitHub Desktop on Arch with `gnome-keyring` installed. Worked flawlessly.

But as soon as I uninstalled `gnome-keyring` GitHub Desktop immediately stops working, even if you already have set up your account and repositories.

Funnily enough, if I reinstall `gnome-keyring` again, GitHub Desktop immediately works again.

![Screenshot_20240405_180447](https://github.com/shiftkey/desktop/assets/115804951/8e3eda83-7783-4be4-b2cb-f22165777ca5)

**Comment by Richardk2n** [comment]
Works for me on both my machines (running Manjaro).
![grafik](https://github.com/shiftkey/desktop/assets/5147221/5a335e8c-22d9-41b0-a21c-79f70fea9289)
Assure, that you have kwallet running and unlocked (If you set a separate password, it is locked by default; Maybe needs to be added to PAM manually on Arch).
Other than that I am not sure, what I might have installed/configured different from you, but I can assure you `gnome-keyring` is not required and thus should not be made a requirement.

**Comment by ghost** [comment]
> Works for me on both my machines (running Manjaro). ![grafik](https://private-user-images.githubusercontent.com/5147221/320205913-5a335e8c-22d9-41b0-a21c-79f70fea9289.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI0NTgwMzgsIm5iZiI6MTcxMjQ1NzczOCwicGF0aCI6Ii81MTQ3MjIxLzMyMDIwNTkxMy01YTMzNWU4Yy0yMmQ5LTQxYjAtYTIxYy03OWY3MGZlYTkyODkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDdUMDI0MjE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGQzOWQyMWY0NmQ1MTkzZTUyZWQzZmFlZjMxNmY2OTMyZTA2ZmYwM2MxNDI0YTI2NTllOTlkMzQwNTllMzZlOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IcKwW0szxDG4_Rbi_vpoPJ8P8DpZR08Fq_9GBr6nXXY) Assure, that you have kwallet running and unlocked (If you set a separate password, it is locked by default; Maybe needs to be added to PAM manually on Arch). Other than that I am not sure, what I might have installed/configured different from you, but I can assure you `gnome-keyring` is not required and thus should not be made a requirement.

Okay, after further digging I noticed that I had installed `kwallet` but actually forgot to install `kwallet-pam` which is important for auto-unlocking of kwallet.

Now it works for me too without `gnome-keyring`.

**Comment by crazymonkyyy** [comment]
> This is a generic DBUS message from the backing store we need to persist credentials, and it's really all the context we see from the application side. I'd suggest checking the `gnome-keyring` keychain is setup.

I have no interest in keyrings and other nonsense; I got this message after disabling kde's on a fresh install

I tried deleting .config/"github desktop", but it still didnt work after trying to log in

**Comment by axzxc1236** [comment]
In my case I have full disk encryption setup and use KDE, I configured SDDM to auto-login (because I already have password required to decrypt boot disk).

Github desktop won't prompt for wallet unlock, but chromium does.

So my work around is to open chromium first, unlock kwallet, then open Github desktop

**Comment by RomanShumikhin** [comment]
Same issue. I use auto-login feature in SDDM, so **kwallet** is not unlocked automatically until I start a browser or another app that that requires it and enter my password, after that GitHub Desktop starts working again.

**Comment by mkurz** [comment]
Does **not** work for me in KDE Plasma 6.2 and using kwallet also.

> [!NOTE]  
> However, instead of "_The name is not activatable_" I get the message "_Can't find session /org/freedesktop/secrets/session/1_"

Never had any problems with kwallet so far, here what's installed:

```sh
$ pacman -Q | grep -iE 'kwallet'
kwallet 6.8.0-1
kwallet-pam 6.2.4-1
kwallet5 5.116.0-1
kwalletcli 3.03-4
kwalletmanager 24.08.3-1
signon-kwallet-extension 24.08.3-1
```

Also, kwallet is a provider of `org.freedesktop.secrets`, see: https://archlinux.org/packages/?q=org.freedesktop.secrets

Only after switching to gnome-keyring things worked...

Digging a bit deeper here is what gets send via dbus when using kwallet, maybe you have an idea?
```
$ dbus-monitor --session
...
method call time=1733857263.408148 sender=:1.121 -> destination=org.freedesktop.secrets serial=11 path=/org/freedesktop/secrets/aliases/default; interface=org.freedesktop.Secret.Collection; member=CreateItem
   array [
      dict entry(
         string "org.freedesktop.Secret.Item.Attributes"
         variant             array [
               dict entry(
                  string "account"
                  string "mkurz"
               )
               dict entry(
                  string "service"
                  string "GitHub - https://api.github.com"
               )
               dict entry(
                  string "xdg:schema"
                  string "org.freedesktop.Secret.Generic"
               )
            ]
      )
      dict entry(
         string "org.freedesktop.Secret.Item.Label"
         variant             string "GitHub - https://api.github.com/mkurz"
      )
   ]
   struct {
      object path "/org/freedesktop/secrets/session/2"
      array of bytes [
         ...<reducted>...
      ]
      array of bytes [
         ...<reducted>...
         ...<reducted>...
         ...<reducted>...
      ]
      string "text/plain"
   }
   boolean true
...
signal time=1733857263.409732 sender=:1.21 -> destination=(null destination) serial=371 path=/org/freedesktop/secrets/collection/kdewallet/0; interface=org.freedesktop.DBus.Properties; member=PropertiesChanged
   string "org.freedesktop.Secret.Item"
   array [
   ]
   array [
   ]
...
signal time=1733857263.409746 sender=:1.21 -> destination=(null destination) serial=372 path=/org/freedesktop/secrets/aliases/default; interface=org.freedesktop.Secret.Collection; member=ItemChanged
   object path "/org/freedesktop/secrets/collection/kdewallet/0"
signal time=1733857263.409749 sender=:1.21 -> destination=(null destination) serial=373 path=/org/freedesktop/secrets/collection/kdewallet; interface=org.freedesktop.Secret.Collection; member=ItemChanged
   object path "/org/freedesktop/secrets/collection/kdewallet/0"
error time=1733857263.409900 sender=:1.21 -> destination=:1.121 error_name=org.qtproject.QtDBus.Error.InvalidObjectPath reply_serial=11
   string "Can't find session /org/freedesktop/secrets/session/2"
...
```

**Comment by mkurz** [comment]
- Related: #940

**Comment by Richardk2n** [comment]
Continues to work perfectly fine with Plasma 6.2 using kwallet for me.
Are you sure your kwallet service is running?

**Comment by mkurz** [comment]
@Richardk2n did you do the inital setup (login to GitHub via browser) with gnome-keyring?

**Comment by Richardk2n** [comment]
I think I did for one of the machines. The other one I am not quite sure but would tend to say no. But I have relogged since and that worked perfectly fine. Pretty sure kwallet is 100% compatible. As you can see from the posts above all issues turn out to be setup/config related. So this issue should really be closes, because it has nothing to do with GitHub desktop.
